internal/convert.go

```diff
@@ -168,27 +168,52 @@ func (conv *Conv) SetDataMode() {
 	conv.mode = dataOnly
 }
 
+func dequeue(queue []string) []string {
+	return queue[1:]
+}
+
 // GetDDL Schema returns the Spanner schema that has been constructed so far.
-// Return DDL in alphabetical table order.
+// We sort DDL in alphabetical order, but we maintain the order of interleaved
+// relation by passing tables through a queue
 func (conv *Conv) GetDDL(c ddl.Config) []string {
 	var tables []string
+	var tablesFk []string
 	for t := range conv.SpSchema {
 		tables = append(tables, t)
+		tablesFk = append(tablesFk, t)
 	}
 	sort.Strings(tables)
 	var ddl []string
-	for _, t := range tables {
-		ddl = append(ddl, conv.SpSchema[t].PrintCreateTable(c))
+	printed := make(map[string]bool)
+	for len(tables) > 0 {
+		t := tables[0]
+		if conv.SpSchema[t].InterleaveInto == "" {
+			ddl = append(ddl, conv.SpSchema[t].PrintCreateTable(conv.SpSchema[t].InterleaveInto, c))
+			printed[t] = true
+			tables = dequeue(tables)
+			continue
+		}
+		if _, found := printed[conv.SpSchema[t].InterleaveInto]; found {
+			ddl = append(ddl, conv.SpSchema[t].PrintCreateTable(conv.SpSchema[t].InterleaveInto, c))
+			printed[t] = true
+			tables = dequeue(tables)
+			continue
+		}
+		tables = dequeue(tables)
+		tables = append(tables, t)
 	}
 
-	// Append foreign key constraints to DDL.
+	// Append foreign key constraints to DDL if table is not interleaved.
```

Comment:
I'm not sure we should do this.
It is perfectly legal to combine foreign key constraints and interleaved tables: just so long as the keys aren't in conflict. For example:

CREATE TABLE t1 (id1 int64, fid int64) PRIMARY KEY (id1);
CREATE TABLE t2 (id1 int64, id2 int64, fid int64) PRIMARY KEY (id1, id2), INTERLEAVE IN PARENT t1 ON DELETE CASCADE;
ALTER TABLE t2 ADD CONSTRAINT fid_constraint FOREIGN KEY (fid) REFERENCES t1 (fid);

Now I realize that in the UI we are potentially converting some foreign key constraints into interleaved tables, but I think the right thing to do there is drop the foreign key constraints that we've converted, not to try to suppress these while printing DDL.